syntax = "proto3";

package Proxy;

service ProxyServer {
    rpc Produce (ProduceRequest) returns (ProduceReply) {}
    rpc Consume (ConsumeRequest) returns (ConsumeReply) {}
    rpc ConsumeAck (ConsumeAckRequest) returns (ConsumeAckReply) {}
}

message ProduceRequest {
    string topic = 1;
    string group = 2;
    string tag = 3;
    string body = 4;
    int32 delayLevel = 5;
}

message ProduceReply {
    int32 code = 1;
    string msg_id = 2;
    string err_msg = 3;
}

message ConsumeRequest
{
    string topic = 1;
    string consumer_group = 2;
}

message ConsumeReply
{
    int32 code = 1;
    string msg_id = 2;
    string body = 3;
    string error_msg = 4;
}

message ConsumeAckRequest
{
    string topic = 1;
    string msg_id = 2;
    string consumer_group = 3;
    int32 status = 4;
}

message ConsumeAckReply
{
    int32 code = 1;
    string msg_id = 2;
    string error_msg = 4;
}